packages/muelu/test/unit_tests/MultiVectorAggregationTransferFactory.cpp

```diff
@@ -0,0 +1,241 @@
+//                    Andrey Prokopenko (aprokop@sandia.gov)
+//                    Ray Tuminaro      (rstumin@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_ScalarTraits.hpp>
+
+
+#include <MueLu_config.hpp>
+
+#include <MueLu_TestHelpers.hpp>
+#include <MueLu_Version.hpp>
+
+#include <Xpetra_MultiVectorFactory.hpp>
+
+#include <MueLu_Utilities.hpp>
+#include <MueLu_CoupledAggregationFactory.hpp>
+#include <MueLu_TentativePFactory.hpp>
+#include <MueLu_TransPFactory.hpp>
+#include <MueLu_FactoryManager.hpp>
+#include <MueLu_MultiVectorTransferFactory.hpp>
+#include <MueLu_MultiVectorAggregationTransferFactory.hpp>
+#include <MueLu_RAPFactory.hpp>
+#include <MueLu_TrilinosSmoother.hpp>
+#include <MueLu_SmootherFactory.hpp>
+#include <MueLu_UncoupledAggregationFactory.hpp>
+
+namespace MueLuTests {
+
+
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(MultiVectorAggregationTransferFactory, Constructor, Scalar, LocalOrdinal, GlobalOrdinal, Node)
+  {
+#   include <MueLu_UseShortNames.hpp>
+    MUELU_TESTING_SET_OSTREAM;
+    MUELU_TESTING_LIMIT_SCOPE(Scalar,GlobalOrdinal,Node);
+    out << "version: " << MueLu::Version() << std::endl;
+
+    RCP<Factory> UncoupledAggFact = rcp(new UncoupledAggregationFactory());
+    RCP<MueLu::MultiVectorAggregationTransferFactory<SC,LO,GO,NO> > mvatf = rcp(new MueLu::MultiVectorAggregationTransferFactory<SC,LO,GO,NO>());
+    mvatf->SetFactory("Aggregates", UncoupledAggFact);
+
+    TEST_EQUALITY(mvatf != Teuchos::null, true);
+  } // Constructor test
+
+  //------------------------------------------------------------------------------------------
+
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(MultiVectorAggregationTransferFactory, Build, Scalar, LocalOrdinal, GlobalOrdinal, Node)
+  {
+#   include <MueLu_UseShortNames.hpp>
+    MUELU_TESTING_SET_OSTREAM;
+    MUELU_TESTING_LIMIT_SCOPE(Scalar,GlobalOrdinal,Node);
+    out << "version: " << MueLu::Version() << std::endl;
+    out << "Tests the action of the transfer factory on a vector." << std::endl;
+    
+    RCP<const Teuchos::Comm<int> > comm = Teuchos::DefaultComm<int>::getComm();
+
+    typedef typename Teuchos::ScalarTraits<SC>::magnitudeType magnitude_type;
+    typedef typename Xpetra::MultiVector<magnitude_type,LO,GO,NO> RealValuedMultiVector;
+
+    Level fineLevel, coarseLevel;
+    TestHelpers::TestFactory<SC,LO,GO,NO>::createTwoLevelHierarchy(fineLevel, coarseLevel);
+    GO nx = 199;
+    RCP<Matrix> A = TestHelpers::TestFactory<SC,LO,GO,NO>::Build1DPoisson(nx);
+    fineLevel.Set("A",A);
+
+    RCP<UncoupledAggregationFactory> aggFact = rcp(new UncoupledAggregationFactory());
+
+    RCP<MultiVector> fineOnes = MultiVectorFactory::Build(A->getRowMap(),1);
+    fineOnes->putScalar(1.0);
+    fineLevel.Set("onesVector",fineOnes);
+
+    RCP<FactoryManager> M = rcp(new FactoryManager());
+    M->SetKokkosRefactor(false);
```

Comment:
So there's no Kokkkos version?

Reply:
This is probably just an artifact of the unit test that I used to construct this. You know, it's a hybrid of both CoordinateTransferFactory and MultiVectorTransferFactory, so it should "support" Kokkos, but since CoordinateTransferFactory transfers using ArrayRCP interfaces, that's the choice I went with. I don't think setting the Kokkos refactor on or off changes anything, but I'll double-check.